gothemis/session/session.go

```diff
@@ -45,8 +74,18 @@ func finalize(ss *SecureSession) {
 }
 
 // New makes a new Secure Session with provided peer ID, private key, and callbacks.
-func New(id []byte, signKey *keys.PrivateKey, callbacks SessionCallbacks) (*SecureSession, error) {
-	ss := &SecureSession{clb: callbacks}
+func New(id []byte, signKey *keys.PrivateKey, callbacks interface{}) (*SecureSession, error) {
```

Comment:
it's bad way to work with `interface{}` instead explicit type. Try to inherit `Callbacks` from `SessionCallbacks` and take value of `Callbacks` type as parameter:
```
type SessionCallbacks interface {
  Callbacks
  GetPublicKeyForId(ss *SecureSession, id []byte) *keys.PublicKey
}
```

Reply:
In this case the old interface `SessionCallbacks` will have to implement the new `GetPublicKeyForID` method which breaks existing code that implements only the old method.

I also don't like these `interface{}` shenanigans, but it seems we have only one viable alternative to this which keeps backwards compatibility:
- introduce a new constructor `ShinyNew` which accepts the new `Callbacks` interface
- keep and deprecate the old `New` constructor which accepts `SessionCallbacks`

New code will be using the `ShinyNew` constructor from now on. Eventually `New` will be removed completely. We can rename `ShinyNew` back into `New` but that's another breaking change, so we'd better come up with a good name from the start. @vixentael, any ideas? 🤔 

(It would be easier if Go supported overloading, but that's not an option.)